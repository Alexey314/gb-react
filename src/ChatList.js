import { List, ListItem, ListItemText } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { useState } from "react";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    maxWidth: 752,
    backgroundColor: theme.palette.background.paper,
  },
  item: {
    backgroundColor: theme.palette.primary.paper,
  },
  itemText: {
    color: theme.palette.text.primary,
  },
}));

function ChatList(props) {
  const classes = useStyles();
  const [selectedId, setSelectedId] = useState("");
  const handleListItemClick = (event, id) => {
    setSelectedId(id);
  };
  const items = props.chatList.map((chat) => {
    return (
      <ListItem
        button
        selected={selectedId === chat.id}
        onClick={(e) => handleListItemClick(e, chat.id)}
        className={classes.item}
        key={chat.id}
      >
        <ListItemText className={classes.itemText} primary={chat.name} />
      </ListItem>
    );
  });
  return <List className={classes.root}>{items}</List>;
}

export default ChatList;
