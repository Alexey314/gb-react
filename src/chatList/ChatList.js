import { List, ListItem, ListItemText } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    // flexGrow: 1,
    // maxWidth: 752,
    // // backgroundColor: theme.palette.grey[100],
  },
  item: {
    backgroundColor: theme.palette.primary.paper,
  },
  itemText: {
    color: theme.palette.text.primary,
  },
}));

function ChatList({ chatList, chatId, onChatSelect }) {
  const classes = useStyles();

  const handleListItemClick = (evt, id) => {
    evt.preventDefault();
    onChatSelect(id);
  };

  const items = chatList.map((chat) => {
    return (
      <ListItem
        button
        selected={chatId === chat.id}
        onClick={(e) => handleListItemClick(e, chat.id)}
        className={classes.item}
        key={chat.id}
      >
        <ListItemText primary={chat.name} />
      </ListItem>
    );
  });

  return (
    <List className={classes.root} component="div">
      {items}
    </List>
  );
}

export default ChatList;
