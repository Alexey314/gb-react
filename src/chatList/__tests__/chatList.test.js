import { render } from "@testing-library/react";
import ChatList from "../ChatList";

describe("chatList", () => {
  it("matches snapshot with empty chat list", () => {
    const chatList = [];
    const chatId = null;
    const onChatSelect = () => {};
    const component = render(
      <ChatList
        chatList={chatList}
        chatId={chatId}
        onChatSelect={onChatSelect}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("matches snapshot with passed chat list", () => {
    const chatList = [
      {
        id: "fkbsdvnzxc",
        name: "Chat 1",
      },
      {
        id: "dfkjbvncklvn",
        name: "Chat 2",
      },
    ];
    const chatId = "fkbsdvnzxc";
    const onChatSelect = () => {};
    const component = render(
      <ChatList
        chatList={chatList}
        chatId={chatId}
        onChatSelect={onChatSelect}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
