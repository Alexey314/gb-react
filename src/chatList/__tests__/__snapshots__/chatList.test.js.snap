// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chatList matches snapshot with empty chat list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiList-root makeStyles-root-1 MuiList-padding"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiList-root makeStyles-root-1 MuiList-padding"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`chatList matches snapshot with passed chat list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiList-root makeStyles-root-4 MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-item-5 MuiListItem-gutters MuiListItem-button Mui-selected"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Chat 1
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root makeStyles-item-5 MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Chat 2
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiList-root makeStyles-root-4 MuiList-padding"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-item-5 MuiListItem-gutters MuiListItem-button Mui-selected"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Chat 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-item-5 MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Chat 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
